### DISCUSSION TITLE
How to generate C-Code header files

### DISCUSSION THREAD

**Discussion by FelixNikolaus-AVT** [SystemRDL/systemrdl-compiler#171] [Q&A] [answered]
Hi,
firstly, this is a grate tool. Thanks a lot to make it open-source. 
Do you have a way to generate header files? I did not see this here until now.

Thanks,

Felix

**Comment by amykyta3** [comment]
I don't have a core PeakRDL tool to generate C headers yet, but I plan to implement one in the future.

In the meantime, it looks like some others in the community have built their own. Perhaps one of these will work for you:
* https://github.com/krcb197/PeakRDL-cheader
* https://github.com/Risto97/PeakRDL-halcpp
* https://github.com/hcadavid/SystemRDL-C-header

**Reply by roowatt** [reply]
> I don't have a core PeakRDL tool to generate C headers yet, but I plan to implement one in the future.

I have been on the lookout for a register tool like PeakRDL for a long time, apart from an expensive commerical tool this looks pretty close to what we need.

However, the project I am working on includes a deeply embedded, bare metal processor. I might be able to some bandwidth from SW dev to help with a tool to generate C headers. My preference would be to work closely with the PeakRDL team and contribute to the core tools.

Is that something you might be interested in and if so, whta would be the next step?

**Comment by amykyta3** [comment, marked as answer]
Updating this discussion thread since it is out of date. PeakRDL indeed supports C header generation now. See https://peakrdl-cheader.readthedocs.io
